tests/components/togrill/test_config_flow.py

```diff
@@ -0,0 +1,121 @@
+"""Test the ToGrill config flow."""
+
+from unittest.mock import Mock
+
+from bleak.exc import BleakError
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant import config_entries
+from homeassistant.components.togrill.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from . import TOGRILL_SERVICE_INFO, TOGRILL_SERVICE_INFO_NO_NAME
+
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_user_selection(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test we can select a device."""
+
+    inject_bluetooth_service_info(hass, TOGRILL_SERVICE_INFO)
+    inject_bluetooth_service_info(hass, TOGRILL_SERVICE_INFO_NO_NAME)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result == snapshot
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={"address": "00000000-0000-0000-0000-000000000001"},
+    )
+    assert result == snapshot
```

Comment:
I would strongly advice against using snapshots here. While snapshots are great for testing large data objects, they make this test hard to read, as in, other config flow tests you can read a test and you know exactly what it does and what is expected, while here you would need to open a separate file and find the right snapshot. Not that having 2 files open is a good reason against this, but I do think that the readability and understandability of test decreases by "hiding" the assertions